types/cjs.d.ts

```diff
@@ -31,6 +31,8 @@ declare namespace megajs {
     inbox: MutableFile
     mounts: MutableFile[]
     files: { [id in string]: MutableFile }
+    filter (query: string | string[] | ((file: File) => void), deep?: boolean): Promise<MutableFile>
```

Comment:
Those don't return promises: since the entire Storage file structure is loaded in memory, those functions work synchronously. Also, they might return `undefined` if a file isn't found. The only argument of the query function is a MutableFile too.

[The implementation of those functions](https://github.com/qgustavor/mega/blob/76cd6e2b29ec91b903c9490186545ca8d46f8798/lib/file.mjs#L427-L485) is quite simple to be fair: they just iterate `file.children` using `Array.prototype.reduce`, using nested calls to handle folders.

(To be fair, I just noticed that the reduce function in navigate could be simplified to `return node?.children && node.find(name)`.)

Reply:
Hmm. I guess we could work on improving the docs a bit to make it more evident. Will address that in future.
You're right about the return type - gets a bit confusing since most things are `any` right now which makes it a bit cumbersome to figure out what goes where.

Reply:
> The only argument of the query function is a MutableFile too.

The Docs read:

```
.find(query, [deep])
Search for files in directores, recursively if the deep argument is true. Queries might be strings (which matches names, case sensitive), arrays of valid names (case sensitive) or a function which receives a File object as argument and should return a boolean.
```

Should we then update the line to read "...a MutableFile object..."? That's how I got File instead of MutableFile for the argument.

Reply:
The docs need to be improved too. When I wrote that I assumed it would be fine, but rewriting this would make it more clear. When I implemented those functions I completely forgot about the TS types...